src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/AzureAttestationBasedEnclaveProvider.NetCoreApp.cs

```diff
@@ -66,35 +66,37 @@ internal class AzureAttestationEnclaveProvider : EnclaveProviderBase
         #region Public methods
         // When overridden in a derived class, looks up an existing enclave session information in the enclave session cache.
         // If the enclave provider doesn't implement enclave session caching, this method is expected to return null in the sqlEnclaveSession parameter.
-        public override void GetEnclaveSession(string servername, string attestationUrl, out SqlEnclaveSession sqlEnclaveSession, out long counter)
+        public override void GetEnclaveSession(string servername, string attestationUrl, bool generateCustomData, out SqlEnclaveSession sqlEnclaveSession, out long counter, out byte[] customData, out long customDataLength)
         {
-            GetEnclaveSessionHelper(servername, attestationUrl, true, out sqlEnclaveSession, out counter);
+            GetEnclaveSessionHelper(servername, attestationUrl, generateCustomData, out sqlEnclaveSession, out counter, out customData, out customDataLength);
         }
 
         // Gets the information that SqlClient subsequently uses to initiate the process of attesting the enclave and to establish a secure session with the enclave.
-        public override SqlEnclaveAttestationParameters GetAttestationParameters()
+        public override SqlEnclaveAttestationParameters GetAttestationParameters(string attestUrl, byte[] customData, long customDataLength)
         {
             ECDiffieHellmanCng clientDHKey = new ECDiffieHellmanCng(DiffieHellmanKeySize);
             clientDHKey.KeyDerivationFunction = ECDiffieHellmanKeyDerivationFunction.Hash;
             clientDHKey.HashAlgorithm = CngAlgorithm.Sha256;
-            byte[] attestationParam = PrepareAttestationParameters();
+            byte[] nonce = customData;
+            int nonceLength = Convert.ToInt32(customDataLength);
```

Comment:
>int nonceLength = Convert.ToInt32(customDataLength); [](start = 12, length = 52)

don't convert from long to int.. pass the long to the PrepareAttestationParameters

Reply:
I assume that this would be `ulong` according to your comment above this one. In PrepareAttestationParameters(), the current serializedNonceLength.Length is 4 for int. Is it OK to change it to 8 if we use `ulong`. 
And I am also thinking about the type of `totalDataSize` 

> // serializing nonce
               byte[] serializedNonce = attestNonce;                               
                byte[] serializedNonceLength = BitConverter.GetBytes(attestNonceLength);

>int totalDataSize = serializedAttestationUrl.Length + serializedAttestationUrlLength.Length + serializedNonce.Length + serializedNonceLength.Length;

1. Should we use `serializedNonce.LongLength` instead of `serializedNonce.Length` ?
2. Or should we use `customDataLength` directly here?
3. If we go with either 1 or 2, totalDataSize needs to be `ulong` or `long`.
4. Same to the `dataCopied` variable.

Reply:
Now i see the problem. If we change the size from 4 to 8, it will trigger an update everywhere. i.e. in sql code, net enclave provider. 
I suggest that lets keep it as int.

Reply:
Yeah, I agree.